File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
 
public class Solution {
    public ListNode MergeTwoLists(ListNode list1, ListNode list2)
{
    ListNode newHead = null;
    ListNode tail = null;

    while (list1 != null && list2 != null)
    {
        ListNode chosen;

        if (list1.val < list2.val)
        {
            chosen = list1;
            list1 = list1.next;
        }
        else
        {
            chosen = list2;
            list2 = list2.next;
        }

        if (newHead == null)
        {
            newHead = chosen;
            tail = chosen;
        }
        else
        {
            tail.next = chosen;
            tail = chosen;
        }
    }

    if (newHead == null)
        return list1 ?? list2;

    tail.next = list1 ?? list2;

    return newHead;
}
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        first  = head
        for _ in range(n): first  = first.next

        dummy = ListNode()
        second = dummy
        second.next = head
        while first:
            second = second.next
            first = first.next

        second.next = second.next.next
        return dummy.next
        
        